### PR TITLE
[Workspace] Add unit tests for inspect page url

### PR DESCRIPTION
**Pull request by yubonluo**
### Description

Add unit tests for inspect page url (https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8718)

## Changelog

- refactor: [Workspace] Add unit tests for inspect page url

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/saved_objects_management/public/utils.ts [hunk 1]
```diff
@@ -12,3 +16,35 @@ export function formatWorkspaceIdParams<
   }
   return others;
 }
+
+export function formatInspectUrl(
+  savedObject: SavedObjectWithMetadata,
+  useUpdatedUX: boolean,
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/utils.ts:22]: We can get useUpdatedUX by using the coreStart, seems we can remove this params.

File: src/plugins/saved_objects_management/public/utils.ts [hunk 2]
```diff
@@ -12,3 +16,35 @@ export function formatWorkspaceIdParams<
...
+  currentWorkspace: WorkspaceObject | null | undefined,
```
Comment (SuZhou-Joe) [inline src/plugins/saved_objects_management/public/utils.ts:23]: We can get currentWorkspace from coreStart as well I suppose.

File: src/plugins/saved_objects_management/public/utils.ts (hunk 1 above)
Comment (yubonluo) [reply src/plugins/saved_objects_management/public/utils.ts:22]: `useUpdatedUX` has been get and called in `saved_objects_table_page.tsx` file, I don't want to get it again.

File: src/plugins/index_pattern_management/public/components/edit_index_pattern/index_header/index_header.test.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        </Provider>
+      </IntlProvider>
+    );
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders the set default button when index is not default and user is in workspace', () => {
```
Comment (SuZhou-Joe) [inline src/plugins/index_pattern_management/public/components/edit_index_pattern/index_header/index_header.test.tsx:72]: Shall we add a case that workspace is not enabled?

**Review by SuZhou-Joe — DISMISSED** [review]

File: src/plugins/index_pattern_management/public/components/edit_index_pattern/index_header/index_header.test.tsx (hunk 3 above)
Comment (yubonluo) [reply src/plugins/index_pattern_management/public/components/edit_index_pattern/index_header/index_header.test.tsx:72]: Sure, I will add it

**Review by Hailong-am — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**